src/radical/pilot/session.py

```diff
@@ -888,7 +900,8 @@ def close(self, **kwargs):
             # terminate all components
             if self._role == self._PRIMARY:
                 self._ctrl_pub.put(rpc.CONTROL_PUBSUB, {'cmd': 'terminate',
-                                                        'arg': None})
+                                                        'arg': None,
+                                                        'fwd': True})
```

Comment:
do we actually need to forward this command? this part is responsible to terminate client-related components, while later, pilot-related (agent side) components are terminated by calling `pmgr.close`. Kinda being precise with calls/commands

Reply:
Alas no: the ticket this PR is addressing stumbled over failed output staging.  That would not be captured by the exit code, the RP task state however does capture that error.

Reply:
There needs to be some cleanup done wrt. agent termination again, and I like to keep this one as a safeguard for now.